Hardware/Hardware/CardReader/CardReaderService.swift

```diff
@@ -45,14 +45,13 @@ public protocol CardReaderService {
 
     /// Creates a PaymentIntent
     /// - Parameter parameters: the intent's parameters
-    func createPaymentIntent(_ parameters: PaymentIntentParameters) -> Future <PaymentIntent, Error>
+    func createPaymentIntent(_ parameters: PaymentIntentParameters) -> Future <Void, Error>
 
-    /// Collects a payment method for the given intent.
-    func collectPaymentMethod(_ intent: PaymentIntent) -> Future<PaymentIntent, Error>
+    /// Collects a payment method.
+    func collectPaymentMethod() -> Future<Void, Error>
 
-    /// Processes a payment after collecting a payment method succeeds.
-    /// - Parameter intent: the payment intent
-    func processPaymentIntent(_ intent: PaymentIntent) -> Future<PaymentIntent, Error>
+    /// Captures a payment after collecting a payment method succeeds.
+    func processPayment() -> Future<PaymentIntent, Error>
```

Comment:
Not sure we need to pass out the entire PI here - we just need the ID as far as we know at this point. If you want to proceed this way, that's fine - I just worry about the surface area making refactoring harder later if we discover we've bound to properties and methods more liberally than we should have.

Reply:
Let's pass the id only, we can always revert later on if we find we need to. I'll update.